EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigi.cc

```diff
@@ -314,17 +317,32 @@ void SiPixelRawToDigi::produce( edm::Event& ev,
     hCPU->Fill( theTimer->realTime() ); 
     hDigi->Fill(formatter.nDigis());
   }
+  
 
+  //------------------------------------
   //send digis and errors back to framework 
+
+  edm::Handle<PixelFEDChannelCollection> pixelFEDChannelCollectionHandle;
+  std::unique_ptr<PixelFEDChannelCollection> PixelFEDChannelCollection_ = nullptr;  
+  if (ev.getByToken(theBadPixelFEDChannelsLabel, pixelFEDChannelCollectionHandle)){
+    
+    const PixelFEDChannelCollection * pfcc= pixelFEDChannelCollectionHandle.product();
+    PixelFEDChannelCollection_ = std::make_unique<PixelFEDChannelCollection>(*pfcc);
+  }
+  
   ev.put(std::move(collection));
   if(includeErrors){
     ev.put(std::move(errorcollection));
     ev.put(std::move(tkerror_detidcollection));
-    ev.put(std::move(usererror_detidcollection), "UserErrorModules");
-    ev.put(std::move(disabled_channelcollection));
+    ev.put(std::move(usererror_detidcollection), "UserErrorModules");      
+    if (PixelFEDChannelCollection_  == nullptr){
+      ev.put(std::move(disabled_channelcollection));
```

Comment:
So, this ```disabled_channelcollection``` is only needed when the ```pixelFEDChannelCollectionHandle``` is not present.
As far as I understood from the discussion we had at the reco meeting, you plan to fully revert this change soon,  and in that case it can temporarily remain as such.
However, if there is the intention of maintaining the DB access of bad channels available also when the "proper implementation" will be integrated in the release, then you could more conveniently compute ```disabled_channelcollection``` (i.e. ```disabledChannelsDetSet```) only conditionally to the existence of that ```pixelFEDChannelCollectionHandle```, i.e. moving L325-L331 before L239

Reply:
We will fully revert these changes,  FEDChannelCollection from the digitizer will not be consumed in SiPixelRawToDigi